test/math_unit_test.hpp

```diff
@@ -90,7 +90,11 @@ bool check_ulp_close(PreciseReal expected1, Real computed, size_t ulps, std::str
     using boost::math::lltrunc;
     // Of course integers can be expected values, and they are exact:
     if (!std::is_integral<PreciseReal>::value) {
-        BOOST_MATH_ASSERT_MSG(!isnan(expected1), "Expected value cannot be a nan.");
+	if (isnan(expected1)) {
```

Comment:
```suggestion
	if (boost::math::isnan(expected1)) {
```

Even the new clang based intel compilers replace `std::isnan` with false.

Reply:
I believe so, but it warns it is going to do it at every optimization level.